### PR TITLE
Book Collection Selector Menu

### PR DESCRIPTION
**Pull request by jakedcolbert**
Implementation of Side By Side and Verse By Verse book collection menus.

### CODE REVIEW

**Comment by jakedcolbert** [comment]
This is what android app does: 
SinglePane: 
- [ ] Highlights selected collection
- [ ] abandons change on cancel and returns to selection before menu was opened.
- [ ] Saves change if ok is hit, regardless of which layout is chosen. Ex. if versebyverse is selected and ok is hit on that tab, any changes made to single pane are also saved.

SideBySide: 
- [ ] Is independent of other layout options, if a collection is selected in singlepane it does not update sidebyside and vice versa. 
- [ ] Changes opposite dropdown to select a new value if selecting a duplicate. Seems to pick the first in order of an array that does not contain the selected collection
- [ ] Saves change if ok is hit, regardless of which layout is chosen. Ex. if versebyverse is selected and ok is hit on that tab, any changes made to single pane are also saved.

VerseByVerse: 
- [ ] Is independent of other layout options, if a collection is selected in singlepane it does not update VerseByVerse and vice versa. 
- [ ] Changes dropdowns that have collections that match the collection selected and ONLY CHANGES THE DUPLICATE TO A NEW VALUE. Seems to pick the first in order of an array that does not contain the selected collection
- [ ] Saves change if ok is hit, regardless of which layout is chosen. Ex. if versebyverse is selected and ok is hit on that tab, any changes made to single pane are also saved.

**Comment by jakedcolbert** [comment]
**Design Proposition:**

There are three files responsible for the implementation of the Collection Menu: 

1) [CollectionList.svelte](https://github.com/sillsdev/appbuilder-pwa/pull/256/files#diff-5bb4ce5d12f8701f335c8d4fa78e3d01484923044da30fb4d51ffd6a311396a9)

**Comment by jakedcolbert** [comment]
Evaluation Checkpoint:

- [ ] Handle cancel requires a $docSets store to overwrite the changes made to $nextDocSet
- [ ] Style bugs: 
> - [ ] Weird dark color highlight on click and hold
> - [ ] Dropdown menus auto open on modal open
> - [ ] occasional appearance of a dark outline on some buttons in modal
- [ ] Scripture App Builder allows duplicate selection in the third VerseByVerse which causes problems with selection logic. Will be avoided if SAB handles duplicates.

**Comment by jakedcolbert** [comment]
Ready for review @chrisvire

**Comment by jakedcolbert** [comment]
Its becoming difficult to keep up with new commits in rebase. The remaining work should be split into two new PRs: 

1. A $ref implementation that accommodates for different layout modes and collections. 
2. Style changes to the CollectionSelector that supports reactivity for the following theme options

Screenshot 2023-07-05 at 3 47 42 PM

**Comment by chrisvire** [comment]
Also need to address #324

File: src/app.d.ts [hunk 1]
```diff
@@ -52,6 +52,19 @@ declare namespace App {
         cells?: GridEntry[];
     }
 
+    interface CollectionEntry {
+        id: string;
+        name: string;
+        singlePane: boolean;
```
Comment (chrisvire) [inline src/app.d.ts:58]: A comment about what `singlePane` means might be helpful. I wasn't sure at first.

File: src/lib/components/CollectionSelector.svelte [hunk 2]
```diff
@@ -23,53 +30,73 @@ Book Collection Selector component.
...
 
+    // ToDo: Set the $refs store to have the docSet using a nextCollection store
     function navigateReference(e) {
+        console.log(e.detail.tab);
         switch (e.detail.tab) {
             case 'Single Pane':
-                docSet = e.detail.text;
-                // force closes active dropdown elements
-                document.activeElement.blur();
+                docSet = $nextDocSet.singlePane.id;
```
[the remark this answers is not in the document]
Comment (chrisvire) [reply src/lib/components/CollectionSelector.svelte:39]: @jakedcolbert I mean use LAYOUT_SINGLE instead of 'Single Pane'

**Comment by jakedcolbert** [comment]
@chrisvire comments addressed. Please see changes.

**Review by chrisvire — APPROVED** [review]
👍

**Pull request merged**